lib/internal/test_runner/test.js

```diff
@@ -570,7 +570,8 @@ class Test extends AsyncResource {
     while (this.pendingSubtests.length > 0 && this.hasConcurrency()) {
       const deferred = ArrayPrototypeShift(this.pendingSubtests);
       const test = deferred.test;
-      test.reporter.dequeue(test.nesting, test.loc, test.name);
+      const type = test.reportedType ?? 'test';
```

Comment:
I think we should make `reportedType` a property on `Test` like it is on `Suite` and drop this.

Reply:
Do you think we should also a reported type on the `TestHook` class since it extends `Test`? I don't think it'll cause any issues, but could lead to future confusion.

Reply:
I set `reportedType` on `TestHook` to `'hook'`.